echopype/utils/io.py

```diff
@@ -151,10 +160,33 @@ def validate_output_path(
     source_file: str,
     engine: str,
     output_storage_options: Dict = {},
-    save_path: Union[None, Path, str] = None,
+    save_path: Optional[Union[Path, str]] = None,
 ) -> str:
     """
-    Assemble output file names and path.
+    Assembles output file names and path.
+
+    The final resulting file will be saved as provided in save path.
+    If a directory path is provided then the final file name will use
+    the same name as the source file and saved within the directory
+    path in `save_path` or echopype's `temp_output` directory.
+
+    Example 1.
+    source_file - test.raw
+    engine - zarr
+    save_path - /path/dir/
+    output is /path/dir/test.zarr
+
+    Example 2.
+    source_file - test.raw
+    engine - zarr
+    save_path - None
+    output is ~/.echopype/temp_output/test.zarr
+
+    Example 3.
+    source_file - test.raw
+    engine - zarr
+    save_path - /path/dir/myzarr.zarr
+    output is /path/dir/myzarr.zarr
```

Comment:
I like these! I think it'll be helpful to add this section to the documentation somewhere.

Reply:
I'm not sure where would be the best place for this @emiliom do you have any suggestion? since you're the most familiar with the docs content. Thanks.

Reply:
I documented this in https://github.com/OSOceanAcoustics/echopype/issues/973#issuecomment-1464721991 so that we can merge this. :)